model/riscv_mem.sail

```diff
@@ -63,6 +63,14 @@ function write_kind_of_flags (aq : bool, rl : bool, con : bool) -> write_kind =
     (true,  false, true)  => throw(Error_not_implemented("sc.aq"))
   }
 
+function is_big_endian(typ : AccessType(ext_access_type)) -> bool = {
+    match effectivePrivilege(typ, mstatus, cur_privilege) {
+      Machine     => bits_to_bool(mstatus[MBE]),
+      Supervisor  => bits_to_bool(mstatus[SBE]),
+      User        => bits_to_bool(mstatus[UBE]),
+    }
+}
+
```

Comment:
Ideally, helper functions like these should not access global state like `mstatus` and `cur_privilege`, and instead receive them as arguments.  This makes them easier to unit test, once we have that infrastructure.

Reply:
Yeah, it's definitely good practice in normal programming, but it seems like Sail really encourages global variables and I think it make it more pseudo-cody.

My plan for dealing with this in unit tests was to re-initialise the model before running each unit test, so although `cur_privilege` is still global you don't have to worry about other tests modifying it, or restoring its state after your test. See [this code](https://github.com/rems-project/sail/pull/928/files#diff-86bc5538625e508bdc56ba5fe9b26858ac5211517533e23f4444255e45aa4136R2244) which I really need to finish!